### ISSUE TITLE
data azure_redis_cache returns 404 for patch schedule

### ISSUE THREAD

**Issue by joshvito** [hashicorp/terraform-provider-azurerm#20488]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.9

### AzureRM Provider Version

3.43.0

### Affected Resource(s)/Data Source(s)

azurerm_redis_cache

### Terraform Configuration Files

```hcl
data "azurerm_redis_cache" "this" {
  name                = "redis-dev"
  resource_group_name = "rg-dev"
}
```

### Debug Output/Panic Output

```shell
╷
│ Error: obtaining patch schedules for Redi (Subscription: "<GUID>"
│ Resource Group Name: "rg-dev"
│ Redis Name: "redis-dev"): patchschedules.PatchSchedulesClient#Get: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="ResourceNotFound" Message="There are no patch schedules found for redis cache 'redis-dev'.\r\nRequestID=<GUID>"
│ 
│   with module.redis_connection.data.azurerm_redis_cache.this,
│   on .terraform/modules/redis_connection/main.tf line 60, in data "azurerm_redis_cache" "this":
│   60: data "azurerm_redis_cache" "this" {
│ 
╵
```

### Expected Behaviour

I expect the data.azurerm_redis_cache to be returned without error. 
This works in provider version 3.42.0 

### Actual Behaviour

Plan command throws the 404 error supplied

### Steps to Reproduce

Create an azurerm_redis_cache resource with **no** patch schedule.
Try to access the data.azurerm_redis_cache in another terraform file. 
The reported error will throw on Plan command.

### References

not at the time of report

**Comment by dave0783** [comment]
We too are experiencing this issue.

**Comment by jhernandez-vendavo** [comment]
I have the same [issue](https://github.com/hashicorp/terraform-provider-azurerm/issues/20489). It appears a [PR](https://github.com/hashicorp/terraform-provider-azurerm/pull/20516) has been submitted to fix it.

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. It looks like this has been resolved as of #20516 As such, I am going to mark this issue as closed.
